pandas/core/generic.py

```diff
@@ -4028,6 +4064,10 @@ def rank(self, axis=0, method='average', numeric_only=None,
         Compute numerical data ranks (1 through n) along axis. Equal values are
         assigned a rank that is the average of the ranks of those values
 
+        Do not call this function via the numpy library, as numpy's version
+        has a completely different purpose (returns number of dimensions in
+        an array)!
+
```

Comment:
How about this warning?  Should it be kept, or should it be the user's responsibility to know the two 'rank' functions serve completely different purposes?